source/tomopy/misc/corr.py

```diff
@@ -441,8 +442,8 @@ def remove_outlier3d(arr, dif, size=3, ncore=None):
     input_type = arr.dtype
     if (input_type != 'float32') and (input_type != 'uint16'):
         arr = dtype.as_float32(arr)  # silent convertion to float32 data type
-    out = np.empty_like(arr)
-
+    out = copy.deepcopy(arr)
```

Comment:
Use `np.copy(arr, order='C')`, instead of making a copy then checking if the copy is contiguous. (If it's not, then you would end up copying the array twice!?)

Reply:
thanks, yes you're right. I guess we could do something like this `np.ascontiguousarray(arr), np.copy(arr, order='C')` in the list of parameters

Reply:
No. We need the reference to the variable `out` so we can return it on line 463.